milli/src/search/criteria/mod.rs

```diff
@@ -20,8 +20,13 @@ mod asc_desc;
 mod proximity;
 pub mod fetcher;
 
+pub struct CriterionContext<'a, 'b> {
+    exclude: &'a RoaringBitmap,
+    word_cache: &'b mut WordDerivationsCache,
+}
+
```

Comment:
This was already implemented before we took this decision. We're gonna need it no matter what, so why not keep it?

Reply:
What I mean is that no criteria use this struct internally, they are just giving it to each other, could you revert the changes related to the `Criterion::next` method you introduced in this PR, this will ease the merge conflicts resolution.